src/libre/ast_analysis.c

```diff
@@ -133,23 +135,23 @@ analysis_iter(struct analysis_env *env, struct ast_expr *n)
 		analysis_iter(env, e);
 
 		if (is_nullable(e)) {
-			set_flags(n, AST_EXPR_FLAG_NULLABLE);
+			set_flags(n, AST_FLAG_NULLABLE);
 		}
 		break;
 	}
 
 	case AST_EXPR_FLAGS:
-		set_flags(n, AST_EXPR_FLAG_NULLABLE); 
-		break;
-
-	case AST_EXPR_CLASS:
-		/* character classes cannot be empty, so they're not nullable */
+		set_flags(n, AST_FLAG_NULLABLE); 
 		break;
 
 	case AST_EXPR_ANCHOR:
 		/* anchor flags will be handled on the second pass */
 		break;
 
+	case AST_EXPR_SUBTRACT:
+		/* XXX: not sure */
+		break;
+
```

Comment:
@silentbicycle ^ this bit here :)

Reply:
I think this is okay, because an empty character class with subtraction is rejected by the parser.

Reply:
Put another way, in this spot this should be handled the same way as LITERAL and NAMED, because it will only transitively contain those as leaf nodes. (Right?)